tests/testthat/test-OTHER-create-track-hub.R

```diff
@@ -0,0 +1,154 @@
+chr10:38,379,045-38,391,967 peakStart bcell kidney
+chr10:38,404,899-38,412,089 peakEnd bcell kidney
+chr10:38,413,073-38,444,133 noPeaks
+"
+
+chrom.sizes.file <- tempfile()
+chrom.sizes <- data.table(chrom="chr10", bases=128616069)
+fwrite(chrom.sizes, chrom.sizes.file, sep="\t", col.names=FALSE)
+repos.url <- "https://raw.githubusercontent.com/tdhock/input-test-data/master/"
+for(bigWig.part in bigWig.part.vec){
+  bigWig.file <- file.path(
+    non.integer.dir, "samples",
+    bigWig.part, "coverage.bigWig")
+  bigWig.url <- paste0(repos.url, bigWig.part, ".bigwig")
+  download.to(bigWig.url, bigWig.file)
+  demo.bigWig <- sub("non-integer", "input", bigWig.file)
+  if(!file.exists(demo.bigWig)){
+    dir.create(dirname(demo.bigWig), showWarnings=FALSE, recursive=TRUE)
+    bw.dt <- readBigWig(bigWig.file, "chr10", 0, 128616069)
+    out.dt <- data.table(chrom="chr10", bw.dt)
+    demo.bedGraph <- sub("bigWig", "bedGraph", demo.bigWig)
+    fwrite(out.dt, demo.bedGraph, sep="\t", col.names=FALSE)
+    cmd <- paste("bedGraphToBigWig", normalizePath(demo.bedGraph), normalizePath(chrom.sizes.file), normalizePath(demo.bigWig))
+    system.or.stop(cmd)
+    unlink(demo.bedGraph)
+  }
+}
+
+
+sample.dir <- dirname(demo.bigWig)
+problem.dir <- file.path(sample.dir, "problems", "chr10:18024675-38818835")
+coverage.bedGraph <- file.path(problem.dir, "coverage.bedGraph")
+unlink(coverage.bedGraph)
+
+for(set.dir in c(non.integer.dir, demo.dir)){
+  labels.file <- file.path(set.dir, "labels", "some_labels.txt")
+  dir.create(dirname(labels.file), showWarnings=FALSE, recursive=TRUE)
+  writeLines(label.txt, labels.file)
+  problems.bed <- file.path(set.dir, "problems.bed")
+  unlink(problems.bed)
+  cat("chr10	18024675	38818835", file=problems.bed)
+}
+
+unlink(non.integer.dir, recursive=TRUE, force=TRUE)
+
+## Set time limit.
+(sample.dir.vec <- Sys.glob(file.path(
+  demo.dir, "samples", "*", "*")))
+prob.dir.vec <- file.path(
+  sample.dir.vec, "problems", "chr10:18024675-38818835")
+limit.dt <- data.table(minutes=5)
+for(prob.dir in prob.dir.vec){
+  dir.create(prob.dir, showWarnings=FALSE, recursive=TRUE)
+  limit.file <- file.path(prob.dir, "target.minutes")
+  fwrite(limit.dt, limit.file, col.names=FALSE)
+}
+
+# Step 0
+convert_labels(demo.dir)
+# Step 1
+problem.dir.glob <- file.path(demo.dir, "samples", "*", "*", "*", "*")
+problem.dir.vec <- Sys.glob(problem.dir.glob)
+for(problem.dir in problem.dir.vec){
+  problem.target(problem.dir)
+}
+# Step 2
+problem.train(demo.dir)
+# Step 3
+prob.dir <- file.path(demo.dir, "problems", "chr10:18024675-38818835")
+problem.pred.cluster.targets(prob.dir)
+# Step 4
+problem.joint.train(demo.dir)
+# Step 5
+job <- file.path(demo.dir, "jobs", "1")
+problem.joint.predict.job(job)
+# Step 6
+plot_all(demo.dir)
+
+url <- paste0("http://CHANGE.THIS/~URL/", basename(demo.dir))
+email <- "toby.hocking@r-project.org"
+genome <- "hg19"
+create_track_hub(demo.dir, url, genome, email)
```

Comment:
can you just run create_track_hub by itself without runnning any of the steps (0-6)?

Reply:
I understand till now for running create_track_hub it requires these files:
  1. `project.dir/samples/*/*/coverage.bigWig`
  2. `project.dir/samples/*/*/joint_peaks.bedGraph or project.dir/samples/*/*/joint_peaks.bigWig`

We can retrieve `coverage.bigWig` from github and `joint_peaks.bigWig/joint_peaks.bedGraph` file by running Pipeline.

If I don't run these steps (0-6), then joint_peaks file will not generate.
Should I hard code the content of `joint_peaks.bigWig/joint_peaks.bedGraph` inside test and write it on file? Or Is there any other way of retrieving the file?

Reply:
would be a simpler test if you just hard code the contents of joint_peaks.bedGraph inside the test.

also you should write one test with no joint_peaks file, and one test with the file. according to https://github.com/tdhock/PeakSegPipeline/blob/master/R/create_track_hub.R#L77 it should work in both cases

Reply:
`create_track_hub` is giving this error when it's called without joint peaks file for a sample. [https://travis-ci.org/tdhock/PeakSegPipeline/jobs/652363486#L3671](https://travis-ci.org/tdhock/PeakSegPipeline/jobs/652363486#L3671)

I think the root cause of this error is because `create_track_hub` is calling `bedGraphToBigWig`[https://github.com/tdhock/PeakSegPipeline/blob/29bff67b0c2b15fb07c2a9eb4581bcac08004f20/R/create_track_hub.R#L75](https://github.com/tdhock/PeakSegPipeline/blob/29bff67b0c2b15fb07c2a9eb4581bcac08004f20/R/create_track_hub.R#L75)

Which is trying to read a bedGraph file without checking it exists or not[https://github.com/tdhock/PeakSegPipeline/blob/29bff67b0c2b15fb07c2a9eb4581bcac08004f20/R/bigwig.R#L15](https://github.com/tdhock/PeakSegPipeline/blob/29bff67b0c2b15fb07c2a9eb4581bcac08004f20/R/bigwig.R#L15)

@tdhock Please look into this and also review the changes.

Reply:
@tdhock I think this [issue](https://travis-ci.org/tdhock/PeakSegPipeline/jobs/652363486#L3671) can be fixed if we add check before reading the bedGraph file Or Is there any other better way to solve this issue?

Reply:
please add `if(file.exists(joint_peaks.bedGraph))` before `bedGraphToBigWig(joint_peaks.bedGraph)` so that it will work even when there are no files.

also please make sure there are two tests.
1. when there are no joint_peaks.bedGraph it should work but the output trackDb file should not have any references to joint_peaks.bigWigs.
2. when there are joint_peaks.bedGraph it should work and the trackDb file should have a line for each joint_peaks.bigWig file.
in both cases there should be a line in the trackDb file for each coverage.bigWig file.

Sorry for all the trouble. But thanks for your persistence. It is good to know that you have good communication / testing skills, and once you are done with this PR, please start working on your GSOC application based on this template https://github.com/rstats-gsoc/gsoc2020/wiki/application-template

Reply:
@tdhock I have implemented the suggested changes.
can you please review it.